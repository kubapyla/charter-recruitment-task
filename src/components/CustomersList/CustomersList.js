import React from 'react';
import { shape, arrayOf, string, number, func } from 'prop-types';
import {
  Table,
  Thead,
  Tbody,
  Tr,
  Th,
  Td,
  Box,
  Heading,
  Text,
  Stack,
  Button,
} from '@chakra-ui/react';
import { Link } from 'react-router-dom';

const CustomersList = ({ customers, generateNewDataset }) => {
  const renderHeader = () => {
    return (
      <Thead>
        <Tr>
          <Th>Customer ID</Th>
          <Th>Customer Name</Th>
          <Th isNumeric>Transactions list</Th>
        </Tr>
      </Thead>
    );
  };

  const renderRows = (customers) =>
    customers.map((customer) => (
      <Tr key={customer.customerId}>
        <Td>{customer.customerId}</Td>
        <Td>{customer.customerName}</Td>
        <Td isNumeric>
          <Button
            colorScheme="blue"
            as={Link}
            to={`/customer/${customer.customerId}`}
          >
            See list
          </Button>
        </Td>
      </Tr>
    ));

  return (
    <>
      <Stack
        as={Box}
        textAlign={'center'}
        spacing={{ base: 8, md: 14 }}
        py={{ base: 8, md: 14 }}
      >
        <Heading
          fontWeight={600}
          fontSize={{ base: '2xl', sm: '4xl', md: '6xl' }}
          lineHeight={'110%'}
        >
          Charter <br />
          <Text as={'span'} color={'green.400'}>
            Recruitment Task
          </Text>
        </Heading>
        <Stack
          direction={'column'}
          spacing={3}
          align={'center'}
          alignSelf={'center'}
          position={'relative'}
        >
          <Button
            colorScheme={'green'}
            bg={'green.400'}
            rounded={'full'}
            px={6}
            _hover={{
              bg: 'green.500',
            }}
            onClick={generateNewDataset}
          >
            Generate new dataset
          </Button>
        </Stack>
      </Stack>
      <Stack
        direction={'column'}
        spacing={3}
        align={'center'}
        alignSelf={'center'}
        position={'relative'}
      >
        <Heading fontWeight={600} lineHeight={'110%'} paddingBottom={10}>
          Customers List
        </Heading>
        <Table variant="simple">
          {renderHeader()}
          <Tbody>{renderRows(customers)}</Tbody>
        </Table>
      </Stack>
    </>
  );
};

CustomersList.propTypes = {
  customers: arrayOf(shape({ customerId: number, customerName: string })),
  generateNewDataset: func.isRequired,
};

CustomersList.defaultProps = {
  customers: [],
};

export default CustomersList;
