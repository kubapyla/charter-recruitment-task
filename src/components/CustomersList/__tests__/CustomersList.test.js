import React from 'react';
import { shallow } from 'enzyme';
import CustomersList from '../CustomersList';

const mockGenerateNewDataset = jest.fn();
const mockCustomers = [
  { customerId: 1, customerName: 'John Smith' },
  { customerId: 2, customerName: 'Jenny Smith' },
];

describe('CustomersList', () => {
  let wrapper;

  it('should match snapshot', () => {
    wrapper = shallow(
      <CustomersList
        customers={mockCustomers}
        generateNewDataset={mockGenerateNewDataset}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
