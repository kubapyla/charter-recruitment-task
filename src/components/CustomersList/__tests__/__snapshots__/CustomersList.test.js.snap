// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomersList should match snapshot 1`] = `
<Fragment>
  <Stack
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "__emotion_base": "div",
        "__emotion_forwardProp": [Function],
        "__emotion_real": [Circular],
        "__emotion_styles": Array [
          [Function],
        ],
        "defaultProps": undefined,
        "render": [Function],
        "withComponent": [Function],
      }
    }
    py={
      Object {
        "base": 8,
        "md": 14,
      }
    }
    spacing={
      Object {
        "base": 8,
        "md": 14,
      }
    }
    textAlign="center"
  >
    <Heading
      fontSize={
        Object {
          "base": "2xl",
          "md": "6xl",
          "sm": "4xl",
        }
      }
      fontWeight={600}
      lineHeight="110%"
    >
      Charter 
      <br />
      <Text
        as="span"
        color="green.400"
      >
        Recruitment Task
      </Text>
    </Heading>
    <Stack
      align="center"
      alignSelf="center"
      direction="column"
      position="relative"
      spacing={3}
    >
      <Button
        _hover={
          Object {
            "bg": "green.500",
          }
        }
        bg="green.400"
        colorScheme="green"
        onClick={[MockFunction]}
        px={6}
        rounded="full"
      >
        Generate new dataset
      </Button>
    </Stack>
  </Stack>
  <Stack
    align="center"
    alignSelf="center"
    direction="column"
    position="relative"
    spacing={3}
  >
    <Heading
      fontWeight={600}
      lineHeight="110%"
      paddingBottom={10}
    >
      Customers List
    </Heading>
    <Table
      variant="simple"
    >
      <ForwardRef>
        <ForwardRef>
          <ForwardRef>
            Customer ID
          </ForwardRef>
          <ForwardRef>
            Customer Name
          </ForwardRef>
          <ForwardRef
            isNumeric={true}
          >
            Transactions list
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
      <ForwardRef>
        <ForwardRef
          key="1"
        >
          <ForwardRef>
            1
          </ForwardRef>
          <ForwardRef>
            John Smith
          </ForwardRef>
          <ForwardRef
            isNumeric={true}
          >
            <Button
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              colorScheme="blue"
              to="/customer/1"
            >
              See list
            </Button>
          </ForwardRef>
        </ForwardRef>
        <ForwardRef
          key="2"
        >
          <ForwardRef>
            2
          </ForwardRef>
          <ForwardRef>
            Jenny Smith
          </ForwardRef>
          <ForwardRef
            isNumeric={true}
          >
            <Button
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              colorScheme="blue"
              to="/customer/2"
            >
              See list
            </Button>
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
    </Table>
  </Stack>
</Fragment>
`;
